Main page 수정

MinKyum-e opened the pull request:
index.html에서 맵을 불러오니까 오류가 생겨서 hooks/useKakaoLoader.js를 추가해서 사용했습니다

- components 추가
   1. Map.js : 마커 fetch 후 카카오맵에 표시
     첫 페이지 로드 및 지도 중심좌표, 지도 줌 레벨 변화시 마커를 fetch 하도록 구현했습니다
  
      (테스트를 위해  위도경도 요청값을 고정시키고 fetch했고 마커 위치도 res 데이터를 약간 수정해서 표시했습니다)
      
   2. SideBar 
      open : 토글 클릭, 마커 클릭 
      close : 토글 클릭, 지도 바닥 클릭

MinKyum-e committed:
Add Kakao Map Component & side bar

MinKyum-e committed:
feat: Update markers on zoom level and map center changes

MinKyum-e committed:
marker fetch 구현

MinKyum-e committed:
clean

DropFL commented on log.txt:
이거 빼주세요

MinKyum-e replied:
아 뺴는걸 깜빡했네요 제거했습니다

DropFL commented on src/components/Map.jsx:
```diff
@@ -0,0 +1,183 @@
+    lat: 37.293976, lng: 126.975059
+  });
+
+  const fetchMarkers = async () => {
```
이거 `transformedMarkers` 만들어서 리턴하는 부분까지만 다른 파일로 분리할 수 있을까요? `api.js` 이런 곳에 만들면 좋을 것 같아요

MinKyum-e replied:
components/api.jsx파일로 옮겼습니다

DropFL replied:
살짝 의미 전달이 잘못 된 것 같아 정정했습니다. 제가 생각했던건 `axios` 를 쓰는 부분부터 `setMarkers` 직전까지의 로직을 분리하는 것이었습니다.

DropFL commented on src/components/Map.jsx:
```diff
@@ -0,0 +1,183 @@
+  };
+
+
+  //페이지 로드시 마커 데이터 가져오기
+  useEffect(() => {
+    fetchMarkers();
+  }, []);
+
+  //드래그, 줌 했을경우 마커 데이터 가져오기
+  useEffect(() => {
+    try {
+      fetchMarkers();
+      console.log('드래그 후 새로운 중심좌표:', mapCenter);
+      console.log('드래그 후 새로운 level:', zoomLevel);
+    } catch (error) {
+      console.error('마커 데이터를 불러오는데 실패했습니다:', error);
+    } finally {
+      setIsLoading(false);
+    }
+  }, [mapCenter, zoomLevel]);
```
위에 있는 `useEffect` 는 필요 없을 것 같네요

MinKyum-e replied:
오 없어도 되네요 삭제했습니다!

DropFL commented on src/components/Map.jsx:
```diff
@@ -0,0 +1,183 @@
+import styled from "styled-components";
+import axios from 'axios';
+
+export default function KakaoMap({ sidebarOpen,sidebarClose, onMarkerClick }) {
```
`sidebarOpen`, `sidebarClose` 는 `onMarkerClick` 내 처리 중 일부로 병합하는게 맞겠습니다.

DropFL commented on src/components/Map.jsx:
```diff
@@ -0,0 +1,183 @@
+
+
+
+  if (!isLoaded || isLoading) {
```
이름이 너무 모호해서, `isSdkLoaded`, `isMarkerLoading` 이 좀 더 좋아보여요
그리고 `isMarkerLoading` 도 그냥 true/false 반전시켜서 `isMarkerLoaded` 로 바꾸는게 좋겠네요

DropFL commented on src/components/Map.jsx:
```diff
@@ -0,0 +1,183 @@
+
+export default function KakaoMap({ sidebarOpen,sidebarClose, onMarkerClick }) {
+  const isLoaded = useKakaoLoader();
+  const [isVisibleId, setIsVisibleId] = useState("");
```
`selectedMarkerId` 라고 합시다

DropFL commented on src/components/Sidebar.jsx:
```diff
@@ -0,0 +1,91 @@
+import styled from "styled-components";
+
+const Sidebar = ({ isSidebarOpen, sidebarClose, sidebarOpen, markerId }) => {
```
Sidebar에 들어갈 내용은 큰 폭으로 달라질 예정이어서, 유연하게 내용을 교체할 수 있도록 `children` 을 받는게 좋을 것 같습니다.

DropFL commented on src/components/Sidebar.jsx:
```diff
@@ -0,0 +1,91 @@
+  cursor: pointer;
+  z-index: 901;
+  
+  &::before {
+    content: '';
+    display: block;
+    width: 0;
+    height: 0;
+    border-top: 12px solid transparent;
+    border-bottom: 12px solid transparent;
+    border-left: 12px solid #666;
+  }
```
사이드바가 열려있냐 닫혀있냐에 따라 화살표 방향이 바뀌어야 합니다.

DropFL commented on src/components/Map.jsx:
```diff
@@ -0,0 +1,183 @@
+      onMarkerClick(id);
+      
+      setTimeout(() => {
+        map.panTo(position, { animate: { cancelable: false } });
```
사이드 바가 열려있는 상태에서 `panTo` 의 결과가 왼쪽으로 쏠리는 현상이 있습니다.
사이드 바에 의해 가려지는 영역을 고려해야 할 것 같아요

DropFL replied:
실제로 지도가 줄어들게 할 수도 있고, 사이드 바의 크기로부터 이동할 거리를 계산해서 `panBy` 를 쓸 수도 있을 것 같네요

DropFL replied:
축척을 고려하여 오프셋을 적용한 `panTo` 를 적용하도록 로직을 바꿨습니다.
그리고 사이드 바의 길이가 픽셀 단위로 고정되는 것이 더 편리해서 400px로 고정했습니다.

MinKyum-e replied:
오 픽셀로하니까 브라우저에서 창 크기 바꿔도 안 삐져나오는군요 자꾸 삐져나와서 transition에 조건 추가했었는데 ㅋㅋ 감사합니다!

DropFL commented on src/pages/Main.jsx:
```diff
@@ -1,25 +1,139 @@
+import KakaoMap from "../components/Map";
 
 function Main() {
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
```
사이드 바는 처음부터 열려있는 것으로 합시다

DropFL commented on package-lock.json:
npm 말고 yarn을 써주세요.

DropFL commented on src/pages/Main.jsx:
```diff
@@ -1,25 +1,139 @@
+      <TopBar>
+        <ProfileContainer>
+          <ProfileButton onClick={() => setIsProfileMenuOpen(!isProfileMenuOpen)}>
+            <img src="/user-icon.svg" alt="프로필" />
```
이 이미지 추가가 안된 것 같네요

MinKyum-e committed:
review 사항 수정

log.txt삭제
map.jsx :
- 68번줄 useEffect삭제
- map, makker load대기 변수 이름 수정
- 사이드바 처음부터 열려있게 수정

main.jsx:
- kakaomap 마커 클릭이벤트처리 통합 (HandleMarkerClick)

sidebar.jsx
- sidebar 열고닫을때 화살표 방향 수정
- sidebar 내용 children으로 받아오게 수정

public/user-icon.svg
- 유저 이미지 추가

MinKyum-e committed:
sidebar onoff 토글로 변경

MinKyum-e committed:
실제 백엔드 마커 fetch

DropFL committed:
Remove package-lock.json

DropFL committed:
Rollback jsconfig.json

DropFL committed:
Disable react/prop-types rule for convenience

DropFL committed:
Format code & remove unused stuffs

DropFL committed:
Add placeholder for sidebar content

DropFL committed:
Set min size of website & fix size of sidebar

DropFL committed:
Adjust position to make the marker on center of visible area

DropFL reviewed (approved):
디테일만 조금 손보고 머지하겠습니다. 고생하셨어요~

MinKyum-e commented:
고생하셨습니다..!
